src/main/java/org/elasticsearch/cluster/metadata/AliasMetaData.java

```diff
@@ -331,7 +349,8 @@ public static AliasMetaData readFrom(StreamInput in) throws IOException {
             if (in.readBoolean()) {
                 searchRouting = in.readString();
             }
-            return new AliasMetaData(alias, filter, indexRouting, searchRouting);
+            AliasFieldsFiltering fieldsFiltering = in.readOptionalStreamable(new AliasFieldsFiltering());
```

Comment:
instead of making this optional all the time can't we write an empty one if it's empty and add a ctor like:

```
public AliasFieldsFiltering(StreamInput in) {
  // read the shit
}
```
so stuff can be final and code is simple?